libc/string_cpy.c

```diff
@@ -0,0 +1,1269 @@
+	TEST_ASSERT_EQUAL_PTR(NULL, memccpy(input, strText, 'x', sizeof(strText)));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(strText, input, sizeof(strText));
+
+#endif
+}
+
+
+TEST(string_memccpy, big)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+#ifndef __phoenix__
+
+	char buff[BIG_NUMB] = { 0 };
+
+	/* Checking capability of handling big blocks of data */
+	TEST_ASSERT_EQUAL_INT(1, big_string());
+	memccpy(buff, long_str, '\xFF', BIG_NUMB);
+	TEST_ASSERT_EQUAL_STRING(buff, long_str);
+	TEST_ASSERT_EQUAL_CHAR(buff[0], '\1');
+	TEST_ASSERT_EQUAL_CHAR(buff[MAX_ASCII_SET / 4], 'A');
+	TEST_ASSERT_EQUAL_CHAR(buff[MAX_ASCII_SET / 2], '\x81');
+
+	/* Offset big chunk of data */
+	TEST_ASSERT_NOT_NULL(memccpy(&buff[128], long_str, '\xFF', BIG_NUMB - MAX_ASCII_SET));
+	TEST_ASSERT_EQUAL_CHAR(buff[0], '\1');
+	TEST_ASSERT_EQUAL_CHAR(buff[MAX_ASCII_SET / 4], 'A');
+	TEST_ASSERT_EQUAL_CHAR(buff[MAX_ASCII_SET / 2], '\x01');
+	TEST_ASSERT_NOT_EQUAL_CHAR(buff[MAX_ASCII_SET / 2], '\x7F');
+	TEST_ASSERT_NOT_EQUAL_CHAR(buff[MAX_ASCII_SET], '\x01');
+	TEST_ASSERT_EQUAL_CHAR(buff[(MAX_ASCII_SET / 4) + 128], 'A');
+	TEST_ASSERT_EQUAL_CHAR(buff[(MAX_ASCII_SET / 2) + 128], '\x81');
+
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+ */
+
+
+TEST_SETUP(string_strncpy)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strncpy)
+{
+}
+
+
+TEST(string_strncpy, basic)
+{
+	int num = 12345678;
+	char payload[MAX_STR_LEN] = { 0 },
+		 buff[MAX_STR_LEN] = { 0 };
+
+	TEST_ASSERT_EQUAL_STRING(TEST_STR1, strncpy(buff, TEST_STR1, sizeof(buff)));
+	TEST_ASSERT_EQUAL_STRING(TEST_STR1, buff);
+
+	/* Looking for leftovers from previous cpy */
+	TEST_ASSERT_EQUAL_STRING(TEST_STR2, strncpy(buff, TEST_STR2, sizeof(buff)));
+	TEST_ASSERT_EQUAL_STRING(TEST_STR2, buff);
+
+	memset(buff, 0, sizeof(buff));
+
+	/* Pragma truncation disable for the length of string tests */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR1, strlen(TEST_STR1) - 9));
+	TEST_ASSERT_EQUAL_STRING("Lorem ip", buff);
+
+	memset(buff, 0, sizeof(buff));
+	sprintf(payload, "%d", num);
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, payload, 5));
+	TEST_ASSERT_EQUAL_STRING("12345", buff);
+
+	memset(buff, 0, sizeof(buff));
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR2, 20));
+	TEST_ASSERT_EQUAL_STRING("Maecenas id commodo", buff);
+
+	/* Combine 2 strings using leftowers in buff */
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR1, 10));
+	TEST_ASSERT_EQUAL_STRING("Lorem ipsud commodo", buff);
+
+#pragma GCC diagnostic pop
+}
+
+
+TEST(string_strncpy, ascii)
+{
+	char buff[MAX_ASCII_SET] = { 0 },
+		 ascii[MAX_ASCII_SET] = { 0 };
+	int i;
+
+	for (i = 1; i < MAX_ASCII_SET; i++) {
+		ascii[i - 1] = i;
+	}
+
+	/* This case test not only ascii set copied but different sizes of copy */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(ascii, strncpy(buff, ascii, i), i);
```

Comment:
You should also check whether buff equals ascii (return value could not be buff)